app/forms/steps/screener/email_consent_form.rb

```diff
@@ -0,0 +1,24 @@
+module Steps
+  module Screener
+    class EmailConsentForm < BaseForm
+      include SingleQuestionForm
+      include HasOneAssociationForm
+
+      has_one_association   :screener_answers
+      yes_no_attribute      :email_consent
+      attribute             :email_address, String, reset_when_no: [:email_consent]
+      validates_presence_of :email_address, if: -> { email_consent&.yes? }
```

Comment:
We have an email validator, so for this, we can do:

```ruby
validates :email_address, email: true, if: -> { email_consent&.yes? }
```

Reply:
good call - done and pushed